File: backend/src/models/Company.ts
import mongoose, { Document, Schema } from 'mongoose';

export interface ICompany extends Document {
  name: string;
  city: string;
  address?: string;
  email?: string;
  phone?: string;
  website?: string;
  linkedIn?: string;
  notes?: string;
  sourceFile?: string;
  createdAt: Date;
  updatedAt: Date;
}

const CompanySchema = new Schema<ICompany>(
  {
    name: {
      type: String,
      required: [true, 'Company name is required'],
      trim: true,
    },
    city: {
      type: String,
      required: [true, 'City is required'],
      trim: true,
      enum: ['Nablus', 'Ramallah', 'Other'],
    },
    address: {
      type: String,
      trim: true,
    },
    email: {
      type: String,
      trim: true,
      lowercase: true,
    },
    phone: {
      type: String,
      trim: true,
    },
    website: {
      type: String,
      trim: true,
    },
    linkedIn: {
      type: String,
      trim: true,
    },
    notes: {
      type: String,
      trim: true,
    },
    sourceFile: {
      type: String,
      trim: true,
    },
  },
  {
    timestamps: true,
  }
);

// Index for faster searching
CompanySchema.index({ city: 1 });
CompanySchema.index({ name: 'text' });

const Company = mongoose.model<ICompany>('Company', CompanySchema);

export default Company;

File: go_version/internal/models/company.go
package models

import "go.mongodb.org/mongo-driver/bson/primitive"

const COMPANIES_COLLECTION = "companies"

type Company struct {
	ID         primitive.ObjectID `bson:"_id,omitempty"`
	Name       string             `bson:"name,omitempty"`
	City       string             `bson:"city,omitempty"`
	Address    string             `bson:"address,omitempty"`
	Email      string             `bson:"email,omitempty"`
	Phone      string             `bson:"phone,omitempty"`
	Website    string             `bson:"website,omitempty"`
	LinkedIn   string             `bson:"linkedIn,omitempty"`
	Notes      string             `bson:"notes,omitempty"`
	SourceFile string             `bson:"sourceFile,omitempty"`
	CreatedAt  primitive.DateTime `bson:"createdAt,omitempty"`
	UpdatedAt  primitive.DateTime `bson:"updatedAt,omitempty"`
	Version    int32              `bson:"__v,omitempty"`
}
